### DISCUSSION TITLE
[security] Session based kong authorizer

### DISCUSSION THREAD

**Discussion by mod** [supabase/supabase#3132] [Feature Requests]
This is a security enhancement,

We are using a similar Authentification mechanism with our open-source project openware/barong
But after long research we have decided to not expose the JWT and keep it behind the gateway.

We expose to the end-user a session-id or a one-time-token,
This allow to have instant and secure logout, we are also adding at the gateway level ip resolution to invalidate session,
blacklist, or maintenance page.

We would like to contribute a golang or LUA Kong plugin, which would be an optional security hardening.
Merging this would benefit both our companies.

![Supabase-authz](https://user-images.githubusercontent.com/80003/132655768-03c40e46-68f3-40d7-a61f-088b0cb7c150.png)

**Comment by kiwicopple** [comment]
Hey @mod - that sounds awesome. It's worth pointing out that we _don't_ have Kong fork so that part would need to be contributed directly to the main Kong org: https://github.com/kong/

For the GoTrue side, we have a fork where we are implementing asymmetric key signing, and this will be exposed on the Dashboard. 
I saw that you opened a Support Ticket for this - we can arrange a call to make sure we're all aligned.

Also - thanks for the sponsorship 🎉
